Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FareCalculator.hs

```diff
@@ -200,7 +201,7 @@ calculateFareParameters params = do
 
     countNightShiftCharge fullRideCost nightShiftCharge = do
       case nightShiftCharge of
-        ProgressiveNightShiftCharge charge -> roundToIntegral $ fromIntegral fullRideCost * charge
+        ProgressiveNightShiftCharge charge -> roundToIntegral $ fromIntegral ((fullRideCost * charge) - fullRideCost)
```

Comment:
This is wrong i believe, charge here should be something like 0.15, which is 15%